llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -5283,9 +5333,6 @@ bool CombinerHelper::matchUDivByConst(MachineInstr &MI) {
   Register Dst = MI.getOperand(0).getReg();
   Register RHS = MI.getOperand(2).getReg();
   LLT DstTy = MRI.getType(Dst);
-  auto *RHSDef = MRI.getVRegDef(RHS);
```

Comment:
Why did this need to move?

Reply:
No it does not. In addition, just like SDIV, the predicate below is what is needed as long as it is exact.

If it is not exact, then it requires a constant or constant vector.

If this isn't true, then SDIV will have to be fixed as well. However, this patch doesn't touch that.

Reply:
Reminder that this is the exact same as SDIV except the shift is logical instead of arithmetic.

Reply:
Doesn't getUnaryPredicate only work on constants and constant vectors?

Reply:
Don't resolve a converstation if you expect someone to read your reply. It causes it to show up collapsed.

Reply:
> Doesn't matchUnaryPredicate fail to match if its not a constant or constant vector?

That is correct. I changed as little as possible while also matching the logic of SDIV.

Reply:
> Doesn't matchUnaryPredicate fail to match if its not a constant or constant vector?

I believe so. I just moved it out of caution because the SDIV version doesn't call the isConstantOrConstantVector method, but UDIV does, and the SDIV version only supports exact division. 

So this was really for the sake of least disruption. However if it is redundant after all, it would have been redundant to begin with.